src/main/java/seedu/address/logic/commands/NoteCommand.java

```diff
@@ -45,14 +47,34 @@ public NoteCommand(Name name, Note note) {
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
-        Person personToEdit = model.findByName(name);
-        if (personToEdit instanceof Maintainer || personToEdit instanceof Staff
-                || personToEdit instanceof Supplier) {
-            throw new CommandException("Person is not of general person class");
+        List<Person> lastShownList = model.getFilteredPersonList();
+
```

Comment:
might be too long, you might want to abstract out what you do in each person class